### PR TITLE
Refactored project build, bumped dependencies

### PR DESCRIPTION
**Pull request by melezov**
Refactored Build.scala
Bumped SBT, Scala versions
Bumped plugins & library dependencies
GitHub is https by default
Removed legacy Eclipse project settings

Also some minor tweaks:
Removed unused import
Removed legacy Scala version switcher for scalatest
Guarded on resolution of credentials file so that its non-existence doesn't spam in SBT console
Added some SBT best practices (`CrossVersion.partialVersion`, `Def.setting`)

### CODE REVIEW

**Comment by sschaef** [comment]
add to whitelist

**Comment by melezov** [comment]
This project configuration would very easily be converted into a multi-project `build.sbt`, as that seems to be the best-practice nowadays (with auto-plugins such as assembly, buildinfo)...

If you wish, I could update the PR with a `build.sbt` definition?

File: project/Build.scala [hunk 1]
```diff
@@ -6,63 +6,59 @@ import scalariform.formatter.preferences._
...
     version := "0.2.0-SNAPSHOT",
-    scalaVersion := "2.11.7",
+    scalaVersion := crossScalaVersions.value.head,
     crossScalaVersions := Seq(
-      "2.11.7",
+      "2.11.8",
       "2.10.6"
     ),
     exportJars := true, // Needed for cli oneJar
-    scalacOptions ++= (scalaBinaryVersion.value match {
```
Comment (sschaef) [inline project/Build.scala]: The `scalacOptions` change is not valid. Scala 2.10 doesn't recognize some options.

**Comment by sschaef** [comment]
Beside of one change, LGTM. If you want you can convert the `.scala` into a `.sbt` one.

**Comment by melezov** [comment]
Done. I took the liberty of splitting it into `build.sbt` and `version.sbt`, my rationale being the assumption that the latter will probably be more frequently updated than the actual build definition.

**Comment by sschaef** [comment]
LGTM. Thanks!

**Pull request merged**